cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/MetadataStore.java

```diff
@@ -203,4 +203,9 @@
    * @return the snapshot of the metadata for entities on or before the given time
    */
   Set<MetadataRecord> getSnapshotBeforeTime(MetadataScope scope,  Set<Id.NamespacedId> entityIds, long timeMillis);
+
+  /**
+   * Upgrades the {@link MetadataStore}
+   */
+  void upgrade();
```

Comment:
can we just have the upgrade method at the dataset level? Store just has business logic and shouldn't be concerned with data storage. Also, you may not have to call this upgrade method explicitly if we just rely on MetadataDataset.upgrade?